### PR TITLE
speed-up time series for points 

### PR DESCRIPTION
**Pull request by AliceBalfanz**
In this PR code changes have been made based on a chat to @forman .
This PR speeds up the time-series extraction for points and adds more information to the xcube-server.log about the duration of time-series extractions. 

EDIT: changed GDAL version dependency (removed restriction of gdal >=2.4,<3.0 , changed to gdal >=2.4 )

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -142,7 +142,8 @@
   ```
 
 ### Enhancements
-
+* time series for points are generated in a faster manner and timers are added to
```
Comment (forman) [inline CHANGES.md]:
```suggestion
* Retrieval of time series in Python API function `xcube.core.timeseries.get_time_series()` has been optimized and is now much faster for point geometries. This enhances time-series performance of `xcube serve`. 
```

File: CHANGES.md [hunk 2]
```diff
@@ -142,7 +142,8 @@
   ```
 
 ### Enhancements
-
+* time series for points are generated in a faster manner and timers are added to 
+  better track performance of time series generation, which now appear in the xcube-server.log
```
Comment (forman) [inline CHANGES.md]:
```suggestion
  * The log-output of `xcube serve` now contains some more details time-series request so performance bottlenecks can be identified more easily from `xcube-serve.log`.
```

File: xcube/core/tile.py [hunk 3]
```diff
@@ -57,8 +57,8 @@ def get_ml_dataset_tile(ml_dataset: MultiLevelDataset,
                         trace_perf: bool = False,
                         exception_type: Type[Exception] = ValueError):
     labels = labels or {}
-    ds_id = hex(id(ml_dataset))
-    image_id = '-'.join(map(str, [ds_id, z, var_name, cmap_name, cmap_range]
+    ds_cache_id = hex(id(ml_dataset))
+    image_id = '-'.join(map(str, [ds_cache_id, z, ml_dataset.ds_id, var_name, cmap_name, cmap_range]
```
Comment (forman) [inline xcube/core/tile.py]: Good!

File: xcube/webapi/controllers/timeseries.py [hunk 4]
```diff
@@ -207,24 +223,29 @@ def get_time_series_for_feature_collection(ctx: ServiceContext,
...
-        except (TypeError, ValueError) as e:
-            raise ServiceBadRequestError("Invalid GeoJSON feature collection") from e
-        shapes.append(geometry)
-    return _get_time_series_for_geometries(dataset, var_name, shapes,
-                                           start_date=start_date,
-                                           end_date=end_date,
-                                           include_count=include_count,
-                                           include_stdev=include_stdev,
-                                           max_valids=max_valids)
+    measure_time = measure_time_cm(disabled=not ctx.trace_perf)
```
Comment (forman) [inline xcube/webapi/controllers/timeseries.py]: I have an idea here, that improves readability of code. Let's chat.

File: xcube/webapi/controllers/timeseries.py [hunk 5]
```diff
@@ -207,24 +223,29 @@ def get_time_series_for_feature_collection(ctx: ServiceContext,
...
+                geometry = shapely.geometry.shape(geometry)
+            except (TypeError, ValueError) as e:
+                raise ServiceBadRequestError("Invalid GeoJSON feature collection") from e
+            shapes.append(geometry)
+        result = _get_time_series_for_geometries(dataset, var_name, shapes,
+                                                 start_date=start_date,
+                                                 end_date=end_date,
+                                                 include_count=include_count,
+                                                 include_stdev=include_stdev,
+                                                 max_valids=max_valids)
```
Comment (forman) [inline xcube/webapi/controllers/timeseries.py]: Code block too large for reasonable performance measures.

File: xcube/webapi/controllers/timeseries.py [hunk 6]
```diff
@@ -31,7 +31,7 @@
 from xcube.core.timecoord import timestamp_to_iso_string
 from xcube.util.geojson import GeoJSON
 from xcube.util.perf import measure_time_cm
-from xcube.webapi.context import ServiceContext
+from xcube.webapi.context import ServiceContext, _LOG
```
Comment (forman) [inline xcube/webapi/controllers/timeseries.py]: I have an idea here, that improves readability of code. Let's chat.

File: xcube/webapi/controllers/timeseries.py [hunk 7]
```diff
@@ -313,11 +334,13 @@ def _collect_ts_result(ts_ds: xr.Dataset,
     else:
         time_indexes = range(num_times)
 
+    values = var.values
```
Comment (forman) [inline xcube/webapi/controllers/timeseries.py]: :)

**Comment by gunbra32** [comment]
Qwi

Norman Fomferra <notifications@github.com> schrieb am Fr., 20. März 2020,
09:36:

**Pull request merged**